forge-gui/res/cardsfolder/f/fumiko_the_lowblood.txt

```diff
@@ -2,7 +2,9 @@ Name:Fumiko the Lowblood
 ManaCost:2 R R
 Types:Legendary Creature Human Samurai
 PT:3/2
-S:Mode$ Continuous | Affected$ Card.Self | AddKeyword$ Bushido:N | CalcKeywordN$ N | Description$ CARDNAME has bushido X, where X is the number of attacking creatures.
-SVar:N:Count$Valid Creature.attacking
+S:Mode$ Continuous | Affected$ Card.Self | AddKeyword$ Bushido:X | Description$ CARDNAME has bushido X, where X is the number of attacking creatures.
+SVar:X:Count$Valid Creature.attacking
+# for Takeno
+SVar:BushidoX:Count$Valid Creature.attacking
```

Comment:
Otherwise it fails this logic here:
https://github.com/Card-Forge/forge/blob/9a5f0b9ff160df858f5ebe9d4203d3e72aa6c6e3/forge-game/src/main/java/forge/game/card/Card.java#L5820-L5827

Reply:
Yes but that's not correct. It would then only trigger as Bushido FixedNumber, so removing another attacker in response doesn't work:
>  X is variable. The bushido bonus is calculated each time Fumiko’s bushido trigger resolves, based on the number of attackers at that time.

>702.1b. An effect that grants an object a keyword ability may define a variable in that ability based on characteristics of that object or other information about the game state. For these abilities, the value of that variable is constantly reevaluated.
Example: Volcano Hellion has the ability "Volcano Hellion has echo {X}, where X is your life total." If your life total is 10 when Volcano Hellion's echo ability triggers but 5 when it resolves, the echo cost to pay is {5}.

Reply:
imo i would be better if KeywordInterface could refer to the StaticAbility/CardState that created it.
(my goal is to make Keywords similar to other CardTraits)

Then the Keyword could use the StateAbility that created it as SVarHolder

also you forgot the SoulShift:N one, then you could have removed CalcKeywordN

also, it would have been better if you had done all this in an extra MR

Reply:
Well, I only meant to fix a few script-related bugs here, so less than 40 lines are still acceptable imo

As for _Kodama of the Center Tree_, it seems to be more complicated and doesn't work with dynamic X
probably because of the dies trigger

Reply:
Because you want too much different things in the same MR

try rebasing this stuff and shrink this MR to a minimum

as for the Keywords stuff, i will try some idea when i get home to have `KeywordInterface` implement `getSVarFallback` via StaticAbility and CardState source